IKZ_plugin/src/ikz_plugin/__init__.py

```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+        type=float,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "liter"},
+        unit="liter",
+    )
+    density = Quantity(
+        type=float,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "gram / liter"},
+        unit="gram / liter",
+    )
+
+
+class MixMaterial(Process, SolutionPreparationStep):
+    """
+    Mix the quantified materials.
+    """
+
+    m_def = Section(
+        a_eln=None,
+        categories=[IKZCategory],
+    )
+    mixing_time = Quantity(
+        type=float,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "second"},
+        unit="second",
+    )
+    temperature = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "celsius"},
+        unit="celsius",
+    )
+    container_type = Quantity(
+        type=str,
+        a_eln={"component": "StringEditQuantity"},
+    )
+    rotation_speed = Quantity(
+        type=float,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "rpm"},
+        unit="rpm",
+    )
+    mixed_material = SubSection(
+        description="""
+        The mixed material.
+        """,
+        section_def=CompositeSystem,
+    )
+
+
+class SolutionPreparationIKZ(Process, EntryData):
+    """
+    Solution preparation class
+    """
+
+    method = Quantity(
+        type=MEnum("Shaker", "Ultrasoncic", "Waiting", "Stirring"),
+        shape=[],
+        a_eln=dict(
+            component="EnumEditQuantity",
+        ),
+    )
+    description = Quantity(
+        type=str,
+        a_eln={"component": "StringEditQuantity"},
+    )
+    atmosphere = Quantity(
+        type=str,
+        a_eln={"component": "StringEditQuantity"},
+    )
+    intended_sample_conc = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
+        unit="mol / liter",
+    )
+    obtained_conc = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
+        unit="mol / liter",
+    )
+    intended_tot_volume = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
+        unit="mol / liter",
+    )
+    obtained_tot_volume = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
+        unit="mol / liter",
+    )
+
+    steps = SubSection(
+        section_def=SolutionPreparationStep,
```

Comment:
So now we can't nest solution preparations but also dropped the alias?

Reply:
true, I will restore the aliases and drop the nesting solution thing